packages/cubejs-api-gateway/src/gateway.ts

```diff
@@ -1195,7 +1195,12 @@ class ApiGateway {
             currentQuery = this.parseMemberExpressionsInQuery(currentQuery);
           }
 
-          const normalizedQuery = normalizeQuery(currentQuery, persistent);
+          let normalizedQuery = normalizeQuery(currentQuery, persistent);
+
+          if (hasExpressionsInQuery) {
```

Comment:
Please, add comment about why early/second call is necessary. I'll probably forget about this whole issue next week :sweat_smile: